src/java/org/apache/cassandra/guardrails/Guardrails.java

```diff
@@ -41,203 +36,214 @@
  */
 public abstract class Guardrails
 {
+    public static final String CUSTOM_GUARDRAILS_FACTORY_PROPERTY = "cassandra.custom_guardrails_factory_class";
+
     private static final GuardrailsConfig config = DatabaseDescriptor.getGuardrailsConfig();
 
+    public static final GuardrailsFactory factory = System.getProperty(CUSTOM_GUARDRAILS_FACTORY_PROPERTY) == null
+                                                    ? new DefaultGuardrailsFactory()
+                                                    : CustomGuardrailsFactory.make(System.getProperty(CUSTOM_GUARDRAILS_FACTORY_PROPERTY));
+
```

Comment:
Also, if I'm not being too dense today, this pattern appears to be very similar to what is used in `CoordinatorClientRequestMetricsProvider`. In that case, it'd be nice sometime if the naming and usage of this pattern could be agreed on and made consistent. Maybe OSS already has a preference, I'm not sure. Just making an observation here fwiw.

Reply:
Ok, yeah, the code seems similar but they are intended to do different things. Nevermind.